src/main/java/slimeknights/tconstruct/tables/recipe/part/PartRecipe.java

```diff
@@ -0,0 +1,99 @@
+package slimeknights.tconstruct.tables.recipe.part;
+
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.tables.TinkerTables;
+
+public class PartRecipe implements IRecipe<IInventory> {
+  protected final ResourceLocation id;
+  protected final String group;
+  protected final ResourceLocation pattern;
+  protected final int cost;
+  protected final ItemStack output;
+
+  public PartRecipe(ResourceLocation id, String group, ResourceLocation pattern, int cost, ItemStack output) {
+    this.id = id;
+    this.group = group;
+    this.pattern = pattern;
+    this.cost = cost;
+    this.output = output;
+  }
+
+  @Override
+  public IRecipeType<?> getType() {
+    return TinkerTables.partRecipeType;
+  }
+
+  @Override
+  public ResourceLocation getId() {
+    return this.id;
+  }
+
+  @Override
+  public ItemStack getIcon() {
+    return new ItemStack(TinkerTables.partBuilder);
+  }
+
+  @Override
+  public IRecipeSerializer<?> getSerializer() {
+    return TinkerTables.partRecipeSerializer.get();
+  }
+
+  /**
+   * Used to check if a recipe matches current crafting inventory
+   */
+  @Override
+  public boolean matches(IInventory inv, World worldIn) {
+    return inv.getStackInSlot(1).getItem() == TinkerTables.pattern.get();
```

Comment:
I don't like the hardcoded slot here, should have the interface grant only the pattern slot. Plus the matches should really check the selected pattern button as another interface return, along with check if the passed material is valid for this part.